lib/src/application.dart

```diff
@@ -109,10 +111,26 @@ class ApplicationConfiguration {
 /// * Register uses and perform various user-related operations through authentication providers
 /// * Synchronize data between the local device and a remote Realm App with Synchronized Realms
 class Application {
-  final AppHandle _handle;
+  late final AppHandle _handle;
   final ApplicationConfiguration configuration;
-
-  Application(this.configuration) : _handle = realmCore.getApp(configuration);
+  late final RealmHttpTransportHandle _httpTransportHandle;
+  // ignore: unused_field
+  late final AppConfigHandle _appConfigHandle;
```

Comment:
I prefer the previous approach, where only the `AppHandle` and `Configuration` is stored, and the `app_config` and `sync_client_config` is created on demand in `realmCore.getApp`.